frontend/integration-tests/tests/kubevirt/utils/utils.ts

```diff
@@ -0,0 +1,205 @@
+  if (leakedArray.length > 0) {
+    console.error(`Leaked ${leakedArray.join()}`);
+    leakedArray.map(r => JSON.parse(r) as {name: string, namespace: string, kind: string})
+      .forEach(({name, namespace, kind}) => {
+        try {
+          execSync(`kubectl delete -n ${namespace} --cascade ${kind} ${name}`);
+        } catch (error) {
+          console.error(`Failed to delete ${kind} ${name}:\n${error}`);
+        }
+      });
+  }
+  leakedResources.clear();
+}
+
+export function addLeakableResource(leakedResources: Set<string>, resource) {
+  leakedResources.add(
+    JSON.stringify({name: resource.metadata.name, namespace: resource.metadata.namespace, kind: resource.kind})
+  );
+}
+
+export function removeLeakableResource(leakedResources: Set<string>, resource) {
+  leakedResources.delete(
+    JSON.stringify({name: resource.metadata.name, namespace: resource.metadata.namespace, kind: resource.kind})
+  );
+}
+
+export function createResource(resource) {
+  execSync(`echo '${JSON.stringify(resource)}' | kubectl apply -f -`);
+}
+
+export function createResources(resources) {
+  resources.forEach(createResource);
+}
+
+export function deleteResource(resource) {
+  const kind = resource.kind === 'NetworkAttachmentDefinition' ? 'net-attach-def' : resource.kind;
+  execSync(`kubectl delete -n ${resource.metadata.namespace} --cascade ${kind} ${resource.metadata.name}`);
+}
+
+export function deleteResources(resources) {
+  resources.forEach(deleteResource);
+}
+
+export async function withResource(resourceSet: Set<string>, resource: any, callback: Function, keepResource: boolean = false) {
+  addLeakableResource(resourceSet, resource);
+  await callback();
+  if (!keepResource) {
+    deleteResource(resource);
+    removeLeakableResource(resourceSet, resource);
+  }
+}
+
+export async function click(elem: ElementFinder, timeout?: number, conditionFunc?: Function) {
+  const _timeout = timeout !== undefined ? timeout : config.jasmineNodeOpts.defaultTimeoutInterval;
+  if (conditionFunc === undefined) {
+    await browser.wait(until.elementToBeClickable(elem), _timeout);
+    await elem.click();
+  } else {
+    do {
+      await browser.wait(until.elementToBeClickable(elem), _timeout);
+      await elem.click();
+    } while (await conditionFunc() === false);
+  }
+}
+
+export async function selectDropdownOption(dropdownId: string, option: string) {
+  await click($(dropdownId));
+  await $(`${dropdownId} + ul`).element(by.linkText(option)).click();
+}
+
+export async function getDropdownOptions(dropdownId: string): Promise<string[]> {
+  const options = [];
+  await $(`${dropdownId} + ul`).$$('li').each(async(element) => {
+    element.getText().then((text) => {
+      options.push(text);
+    });
+  });
+  return options;
+}
+
+export async function fillInput(elem: ElementFinder, value: string) {
+  // Sometimes there seems to be an issue with clear() method not clearing the input
```

Comment:
This seems really hacky. We should get to the bottom of why this is an issue instead of adding arbitrary retries